packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -248,89 +298,106 @@ export class HostedPluginSupport {
                 hostContributions.set(host, dynamicContributions);
                 toDisconnect.push(Disposable.create(() => {
                     contributions!.state = PluginContributions.State.LOADED;
-                    console.log(`[${plugin.model.id}]: Disconnected.`);
+                    console.log(`[${this.watcher.clientId}][${pluginId}]: Disconnected.`);
                 }));
             }
         }
-        for (const pluginId of toUnload) {
-            const contribution = this.contributions.get(pluginId);
-            if (contribution) {
-                contribution.dispose();
-            }
-        }
-        if (loaded || toUnload.size) {
-            this.onDidChangePluginsEmitter.fire(undefined);
-        }
         return hostContributions;
     }
 
-    protected startPlugins(
-        contributionsByHost: Map<PluginHost, PluginContributions[]>,
-        initData: PluginsInitializationData,
-        container: interfaces.Container
-    ): Disposable {
-        const toDisconnect = new DisposableCollection();
+    protected async startPlugins(contributionsByHost: Map<PluginHost, PluginContributions[]>, toDisconnect: DisposableCollection): Promise<void> {
+        const [hostLogPath, hostStoragePath] = await Promise.all([
+            this.pluginPathsService.getHostLogPath(),
+            this.getStoragePath()
+        ]);
+        if (toDisconnect.disposed) {
+            return;
+        }
+        const configStorage = { hostLogPath, hostStoragePath };
+        const activationEvents = [...this.activationEvents];
```

Comment:
move it after obtaining a manager, since in the meantime more activation events can happen